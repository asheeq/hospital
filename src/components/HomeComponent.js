import React, { Component } from 'react'
import { Link } from 'react-router-dom'

class WelcomeComponent extends Component {
    render() {
        return (
            <>
                <div className="container">
                    <div className="main-title">
                    <h1>Welcome to Spring Hospital.</h1>
                    <p>Spring Hospital is one of the leading hospitals in Bangladesh.</p>
                    <h3>Want an appointment? Click <Link to="/department">here</Link>.</h3>
                    </div>
                </div>
            </>
        )
    }
}
export default WelcomeComponent
