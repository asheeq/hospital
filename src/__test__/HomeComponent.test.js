import React from "react";
import { shallow } from "enzyme";
import HomeComponent from '../components/HomeComponent.js'
import { Link } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

it("Renders without crashing", () => {
    shallow(<HomeComponent />);
});

test('Link matches snapshot', () => {
  const component = renderer.create(
    <StaticRouter>
      <Link to="/department" />
    </StaticRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});