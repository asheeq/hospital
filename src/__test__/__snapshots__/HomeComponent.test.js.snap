// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link matches snapshot 1`] = `
<a
  href="/department"
  onClick={[Function]}
/>
`;
